src/ray/core_worker/reference_count.cc

```diff
@@ -239,6 +239,36 @@ void ReferenceCounter::AddDynamicReturn(const ObjectID &object_id,
   AddNestedObjectIdsInternal(generator_id, {object_id}, owner_address);
 }
 
+void ReferenceCounter::OwnDynamicStreamingTaskReturnRef(const ObjectID &object_id,
+                                                        const ObjectID &generator_id) {
+  absl::MutexLock lock(&mutex_);
+  // NOTE: The upper layer (the layer that manges the object ref stream)
+  // should make sure the generator ref is not GC'ed when the
+  //
+  auto outer_it = object_id_refs_.find(generator_id);
+  if (outer_it == object_id_refs_.end()) {
+    // Generator object already went out of scope.
```

Comment:
add warning?

Reply:
Actually I think we don't need a warning here. It can happen under normal circumstance (e.g., generator went out of scope before task finishes). I instead add a debug message!